noop/shared/src/main/scala/org/typelevel/log4cats/noop/NoOpFactory.scala

```diff
@@ -22,13 +22,19 @@ import cats.Applicative
 object NoOpFactory extends LoggerFactoryGenCompanion {
   def apply[F[_]: Applicative]: LoggerFactory[F] = impl[F]
 
-  def impl[F[_]](implicit F: Applicative[F]): LoggerFactory[F] = new LoggerFactory[F] {
-    override def getLoggerFromName(name: String): SelfAwareStructuredLogger[F] = {
-      val _ = name
-      NoOpLogger.impl[F]
+  def strictEvalArgs[F[_]: Applicative]: LoggerFactory[F] = impl_[F](evaluateArgs = true)
```

Comment:
How are people planning to choose this?  An effectful toggle on a CI flag?

Reply:
As for me, the code under test receives an implicit  `Logger` or `LoggerFactory` parameter. So it is the test setup code that would create that `strictEvalArgs` instance.

Reply:
If this noop-things, why don't make them a default option?

Reply:
> If this noop-things, why don't make them a default option?

I see 2 reasons not to do it:

- It would change the current behaviour. Existing user would surely do not like to see their trace message becoming systematically evaluated in production.
- NoOp  means doing nothing. Evaluating args deviate a bit from that and should not be the default IMO